### DISCUSSION TITLE
Features request

### DISCUSSION THREAD

**Discussion by nico7550** [86Box/86Box#2918] [Ideas]
Hi,

First, thank for 86box, I was waiting for year to have such tools to fulfil my dos needs :-) 

If possible can you please think about the next possible features:

-pause/save/reload state
-video/screen buffer (when text only) recording
-buttons (power/reset/turbo/volume (if sound card choosen)) toolbar
-ram slider in settings
-select the processor before the motherboard
-intregration of 86box manager and PCemufileinjector
-copy/paste exchange for text only
-fake beige screen border around windows for a more immersive experience (full screen option)
-screen type in option (hercule, cga, ega, vga, svga...)

And a speciale request: I discover my favorite GPU brand release a last one before dying...
ET6100
https://www.ebay.ca/itm/IMAGRAPH-CORP-SPECTRUM-FRAME-GRABBER-CARD-Tseng-Labs-ET6100-VGA-Card-PCI-Slot/324284500683?hash=item4b80dccacb:g:YKMAAOSwUwRfUo2-

Thanks

Hope it wasn't boring

**Comment by dhrdlicka** [comment]
> -pause/save/reload state

Pause is already there, for save/reload state there is #955 already open.

> -video/screen buffer (when text only) recording

Not sure what you mean by that.

> -buttons (power/reset/turbo/volume (if sound card choosen)) toolbar

This is planned for a future release. (#827)

> -ram slider in settings

I am not quite sure what would be the advantage as compared to the current text box.

> -select the processor before the motherboard

The latest v3.0 builds have a "machine type" selection before the motherboard mostly based on socket types, which I think is a 
more elegant solution than merely choosing the CPU before the motherboard.

> -intregration of 86box manager and PCemufileinjector

Seamless generation of floppy/CD images, which would pretty much implement the same functionality in the emulator itself, is 
definitely a thing we want to look at in the future. (#187)

> -copy/paste exchange for text only

The emulator can't possibly know what character set and keyboard layout the VM is using, so this would likely be more or less 
broken for anything that's not CP437 and US keyboard layout.

> -fake beige screen border around windows for a more immersive experience (full screen option)

I guess this could be added as an option.

> -screen type in option (hercule, cga, ega, vga, svga...)

I am not sure what effect that should have on the resulting screen output. Some cards already do have a screen type option in device configuration for selecting monochrome color, if that is what you mean.

**Comment by nico7550** [comment]
-video/screen buffer (when text only) recording
Not sure what you mean by that.
==> the idea is to ouput to text file the screen text like a text video recorder

    -ram slider in settings
I am not quite sure what would be the advantage as compared to the current text box.
==> more comfort :-)

Thanks for your answer

**Comment by usergithub64** [comment]
> -fake beige screen border around windows for a more immersive experience (full screen option)

This is exactly what you don't need to do, as the fake beige screen border will get in the way in full screen mode. When using full screen mode, the fake beige screen border would greatly interfere with the full immersion in the work or game process.

**Comment by OBattler** [comment]
> the idea is to ouput to text file the screen text like a text video recorder

The problem with that is, that it would need a specific handler for each graphics card, which would have to be aware of the mode and memory mappings currently set, and how everything is organized. It would be more complicated than you think it is.

> more comfort :-)

I personally find a combo box with a spinner more comfortable as I can set the RAM amount precisely, which is going to be a problem with a slider.

**Comment by overthetop78** [comment]
Hello, it would be nice to be able to do a full screen mode of 86box (not the screen output of the emulation) where we would have a screen and below or next to it, depending on the model, the tower / desktop. On it, we could see the CD drive, the Floppy drive, ZIP and MO if existing and be able to interact with it to change the disk image. Also HDD and Power LEDs. As well as the Floppy, CD, MO and ZIP LEDs.
it could give a better "old-machine" visual.
It is purely aesthetic but it allows a better immersion.

**Comment by explorerdotexe** [comment]
> 
> 
> Hello, it would be nice to be able to do a full screen mode of 86box (not the screen output of the emulation) where we would have a screen and below or next to it, depending on the model, the tower / desktop. On it, we could see the CD drive, the Floppy drive, ZIP and MO if existing and be able to interact with it to change the disk image. Also HDD and Power LEDs. As well as the Floppy, CD, MO and ZIP LEDs.
> it could give a better "old-machine" visual.
> It is purely aesthetic but it allows a better immersion.

That's actually an idea i had as well, even though it is pretty useless from an emulation point of view. There are projects like PCbi which are being worked on, however that's only for PCem.

**Comment by overthetop78** [comment]
> > Hello, it would be nice to be able to do a full screen mode of 86box (not the screen output of the emulation) where we would have a screen and below or next to it, depending on the model, the tower / desktop. On it, we could see the CD drive, the Floppy drive, ZIP and MO if existing and be able to interact with it to change the disk image. Also HDD and Power LEDs. As well as the Floppy, CD, MO and ZIP LEDs.
> > it could give a better "old-machine" visual.
> > It is purely aesthetic but it allows a better immersion.
> 
> That's actually an idea i had as well, even though it is pretty useless from an emulation point of view. There are projects like PCbi which are being worked on, however that's only for PCem.

I saw what PCBi gave but I think it would be better fixed rather than 3D. It would make, in my opinion, too "video game". A fixed display would already be easier to set up.

**Comment by explorerdotexe** [comment]
> > > Hello, it would be nice to be able to do a full screen mode of 86box (not the screen output of the emulation) where we would have a screen and below or next to it, depending on the model, the tower / desktop. On it, we could see the CD drive, the Floppy drive, ZIP and MO if existing and be able to interact with it to change the disk image. Also HDD and Power LEDs. As well as the Floppy, CD, MO and ZIP LEDs.
> > > it could give a better "old-machine" visual.
> > > It is purely aesthetic but it allows a better immersion.
> > 
> > 
> > That's actually an idea i had as well, even though it is pretty useless from an emulation point of view. There are projects like PCbi which are being worked on, however that's only for PCem.
> 
> I saw what PCBi gave but I think it would be better fixed rather than 3D. It would make, in my opinion, too "video game". A fixed display would already be easier to set up.

Considering how intensive 86box can be, if this feature were to be implemented it should be a still image. Maybe there could even be an option to view it in a limited color palette, dithering and all, for that 90’s PC game vibe.

**Comment by dhrdlicka** [comment]
Continue this discussion under issue #1273, please.
